internal/stats/aggregate.go

```diff
@@ -0,0 +1,212 @@
+package stats
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type AggregationQuery struct {
+	EnableAggregation bool     `json:"enable-aggregation,omitempty"`
+	GroupByFields     []string `json:"group-by,omitempty"`
+	AggregateField    string   `json:"aggregate,omitempty"`
+}
+
+type QueryResponse struct {
+	Response []map[string]any `json:"response"`
+}
+
+const (
+	COUNT_KEY  = "count"
+	MAX_KEY    = "max"
+	MEAN_KEY   = "mean"
+	MEDIAN_KEY = "median"
+	MIN_KEY    = "min"
```

Comment:
These seem strange.
Why are we pulling the values out by key instead of just putting the result (util.AggregateValues) directly in our result?